cookbooks/cdo-varnish/libraries/http_cache.rb

```diff
@@ -69,12 +69,21 @@ def self.config(env)
             proxy: 'dashboard',
             headers: LANGUAGE_HEADER,
             cookies: whitelisted_cookies
+          },
+          {
+            path: %w(
+              /
+              /learn*
```

Comment:
LGTM after the few remaining comments are addressed.  My one big remaining
concern is whether no language cookies is the right default for Pegasus--
seems like we would want to default to including language headers to fail
safe.  As we saw with starwars, it's frequently not obvious in testing that
the headers are being stripped incorrectly for new content because the
content has been localized, so it's easy for a bug to go to production.

On Wed, Nov 11, 2015 at 2:34 PM, Will Jordan <notifications@github.com>
wrote:

> In cookbooks/cdo-varnish/libraries/http_cache.rb
> <https://github.com/code-dot-org/code-dot-org/pull/5246#discussion_r44597094>
> :
>
> > @@ -69,12 +69,21 @@ def self.config(env)
> >              proxy: 'dashboard',
> >              headers: LANGUAGE_HEADER,
> >              cookies: whitelisted_cookies
> > +          },
> > +          {
> > +            path: %w(
> > +              /
> > +              /learn*
>
> If we made that change, I would worry that /learn/* would not match
> against /learn, which would be a bigger problem since that's the base
> index-path that e.g. code.org/learn resolves to.
>
> I'll need to do some targeted testing through CloudFront to be absolutely
> sure on the behavior for this edge-case. Worse-case, we would need to list
> two separate behaviors with path-patterns /learn and /learn/* in order to
> conservatively match only the specified path root and all of its children
> paths. (For this, we would need our distribution limits increased - I'll
> actually put in a request for anyway, since it's likely we'll need more
> than 25 path-specific behaviors at some point in the future regardless.)
>
> I think sorting this out is low-priority until we actually need to use a
> path where different cache behaviors are needed - as far as I know we don't
> have any problematic paths like /learn-french in our system right now so
> things should work fine as-is until we do.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/5246/files#r44597094>.
>